src/optimizer/rules/rules.py

```diff
@@ -300,7 +300,7 @@ def _rotate(self, before):
             node_cost += cost
         children_cost = sorted(children_cost, key=lambda entry: entry[0])
         updated_children = [entry[1] for entry in children_cost]
-        before.chidren = updated_children
+        before.rewrite_children(updated_children)
```

Comment:
@xzdandy @gaurav274 
Please check my fix here.
First, it is a typo. Second, `children` is a property so it does not get override in place.
My fix is to add a rewrite function for abstract expression, so reordering of children can happen in place.

Reply:
https://github.com/georgia-tech-db/eva/pull/136/commits/f60fbe7e9be34834b5cd574e836b86a30f1f0cf0#diff-4579d676d8e267e19140e82e669831b8238ef971eedb783b0ef07fcd310b88d4R88

My fix for expression is here.

Reply:
This is an experimental rule in EVA reuse work (and not enabled I believe in the mat branch). I would suggest removing it from the initial optimize merge.  In practice, you use the following way to make a property updatable.
```
@children.setter
def children(self, value):
    self._children = value
```

Reply:
Ok, then I will just comment out the entire rule for now?

Reply:
Actually this looks like a very useful rule to me. Can we just leave here? I will revert it back to original version but add a setter to AbstractExpression.

Reply:
We have not tested this rule as the version in the mat branch since I do not expect it in the mat branch. That's why there exists a typo error and the property problem you found.
If we want to leave it in the pr, we probably should test it first to make sure it actually works.
And I would recommend adding the following comments to the Rule to avoid confusion:
1. The rule is not enabled.
2. The rule is not functional without pr #138.

Reply:
Yup, comment out this rule. I'm replacing this with another more generic rule (UDF reordering).